### PR TITLE
Add option for hints to skip probing the same client

### PR DESCRIPTION
**Pull request by jerryz123**
Hints from the client that owns a line should not probe the client.
Add an option to disable same-client probes from hints

### CODE REVIEW

**Review by sequencer — COMMENTED** [review]
I don't have enough confidence in approving this, only looking at the code change.
In the future, I do think add coverage and add CI is important for such features improvement. 
However since this configuration is set to false by default, so it's still safe for other users. but ask for two more days to let me read the code base again. ;p

File: design/craft/inclusivecache/src/Parameters.scala [hunk 1]
```diff
@@ -267,12 +268,12 @@ object MetaData
     ((opcode === TLMessages.AcquireBlock || opcode === TLMessages.AcquirePerm) && param =/= TLPermissions.NtoB)
   }
   // Does a request prove the client need not be probed?
-  def skipProbeN(opcode: UInt): Bool = {
+  def skipProbeN(opcode: UInt, hintsSkipProbe: Boolean): Bool = {
     // Acquire(toB) and Get => is N, so no probe
     // Acquire(*toT) => is N or B, but need T, so no probe
     // Hint => could be anything, so probe IS needed
```
Comment (sequencer) [inline design/craft/inclusivecache/src/Parameters.scala:274]: This comment need to be edited:
```suggestion
    // Hint => could be anything, so probe IS needed, if hintsSkipProbe is enabled, skip probe the same client
```

**Review by sequencer — APPROVED** [review]
need to add some comments.
It seems that this function does provide a performance improvement to some intensive prefetch designs, the design of "always probe hint" slow down the uarch which prefetch unit and cache are highly decoupled to each other.

**Comment by jerryz123** [comment]
I have some confidence enabling the flag does not cause functionality issues. After all, hints can always be ignored by the implementation.

**Pull request merged**